src/test/java/io/jenkins/plugins/analysis/warnings/recorder/SnippetGeneratorITest.java

```diff
@@ -0,0 +1,159 @@
+                .setAggregatingResults(false)
+                .setBlameDisabled(false)
+                .setEnabledForFailure(false)
+                .setIgnoreFailedBuilds(true)
+                .setIgnoreQualityGate(false)
+                .setPattern("", 1)
+                .setReferenceJobName("")
+                .setSourceCodeEncoding("");
+
+        SnippetGenerator generator = (SnippetGenerator) config;
+        String script = generator.generateScript();
+
+        assertThat(script).isEqualTo("recordIssues(tools: [java()])");
+        assertThat(script).contains("recordIssues");
+        assertThat(script).contains("tools: [java()]");
+    }
+
+    /**
+     * Tests the configuration of recordIssues that differs most from the default configuration.
+     */
+    @Test
+    public void antiDefaultConfigurationExplicitTest() {
+        WorkflowJob job = createPipeline();
+        FreestyleConfiguration config = new SnippetGenerator(
+                getWebPage(JavaScriptSupport.JS_ENABLED, job, "pipeline-syntax/"))
+                .selectRecordIssues().setTool("Java")
+                .setAggregatingResults(true)
+                .setBlameDisabled(true)
+                .setEnabledForFailure(true)
+                //.setHealthReport(null,null,Severity.WARNING_LOW)  //default int not possible
+                .setIgnoreFailedBuilds(false)
+                .setIgnoreQualityGate(true)
+                .setPattern("firstText", 1)
+                .setReferenceJobName("someText")
+                .setSourceCodeEncoding("otherText");
+
+        SnippetGenerator generator = (SnippetGenerator) config;
+        String script = generator.generateScript();
+
+        assertThat(script).contains("recordIssues");
+        assertThat(script).contains("aggregatingResults: true");
+        assertThat(script).contains("blameDisabled: true");
+        assertThat(script).contains("enabledForFailure: true");
+        assertThat(script).contains("ignoreFailedBuilds: false");
+        assertThat(script).contains("ignoreQualityGate: true");
+
+        assertThat(script).contains("pattern: 'firstText'");
+        assertThat(script).contains("referenceJobName: 'someText'");
+        assertThat(script).contains("sourceCodeEncoding: 'otherText'");
+        assertThat(script).contains("tools: [java(");
+        assertThat(script).contains(")]");
+    }
+
+    /**
+     * Tests the HealthReport.
+     */
+    @Test
+    public void configureHealthReportTest() {
+        WorkflowJob job = createPipeline();
+        FreestyleConfiguration config = new SnippetGenerator(
+                getWebPage(JavaScriptSupport.JS_ENABLED, job, "pipeline-syntax/"))
+                .selectRecordIssues().setTool("Java")
+                .setHealthReport(1, 9, Severity.WARNING_LOW);
+
+        SnippetGenerator generator = (SnippetGenerator) config;
+        String script = generator.generateScript();
+
+        assertThat(script).contains("recordIssues");
```

Comment:
I think here it makes more sense to test for the exact string (rather than 4 assertions).